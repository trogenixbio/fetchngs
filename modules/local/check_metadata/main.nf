process CHECK_METADATA {
    tag "$id"
    label 'error_retry'

    conda "conda-forge::python=3.9.5"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/python:3.9--1' :
        'biocontainers/python:3.9--1' }"

    input:
    path metadata_json
    path metadata_schema

    output:
    path "valid.metadata.json"  , emit: json
    path "versions.yml"         , emit: versions

    script:
    """
    check_schema.py \\
        ${metadata_json} \\
        ${metadata_schema}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //g')
    END_VERSIONS
    """
}
