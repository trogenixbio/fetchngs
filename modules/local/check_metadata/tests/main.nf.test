nextflow_process {

    name "Test Process CHECK_METADATA"
    script "../main.nf"
    process "CHECK_METADATA"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = file("$baseDir" + "/" + params.modules_testdata_base_path_local + "metadata.json", checkIfExists: true)
                input[1] = file("$baseDir" + "/" + params.modules_testdata_base_path_local + "ERC000011-TRGX01.json", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
