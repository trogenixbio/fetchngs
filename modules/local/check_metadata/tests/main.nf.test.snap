{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "validation_report.txt:md5,0740e7a3cceb8a1e8b96ea634ff3c508"
                ],
                "1": [
                    "versions.yml:md5,909fbbdae9a5e1ed8f86453b72c2e236"
                ],
                "json": [
                    "validation_report.txt:md5,0740e7a3cceb8a1e8b96ea634ff3c508"
                ],
                "versions": [
                    "versions.yml:md5,909fbbdae9a5e1ed8f86453b72c2e236"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-22T10:04:59.712276004"
    }
}